beacon-chain/rpc/eth/beacon/handlers_pool.go

```diff
@@ -0,0 +1,174 @@
+package beacon
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/go-playground/validator/v10"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/core/feed"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/core/feed/operation"
+	corehelpers "github.com/prysmaticlabs/prysm/v4/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/rpc/eth/shared"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/primitives"
+	"github.com/prysmaticlabs/prysm/v4/crypto/bls"
+	http2 "github.com/prysmaticlabs/prysm/v4/network/http"
+	ethpbalpha "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/v4/time/slots"
+	"go.opencensus.io/trace"
+)
+
+// ListAttestations retrieves attestations known by the node but
+// not necessarily incorporated into any block. Allows filtering by committee index or slot.
+func (s *Server) ListAttestations(w http.ResponseWriter, r *http.Request) {
+	_, span := trace.StartSpan(r.Context(), "beacon.ListAttestations")
+	defer span.End()
+
+	rawSlot := r.URL.Query().Get("slot")
+	var slot uint64
+	if rawSlot != "" {
+		var valid bool
+		slot, valid = shared.ValidateUint(w, "Slot", rawSlot)
+		if !valid {
+			return
+		}
+	}
+	rawCommitteeIndex := r.URL.Query().Get("committee_index")
```

Comment:
we can probably even make this whole thing a helper function , seems like slot and raw committee index do the same thing?